src/box/memtx_hash.cc

```diff
@@ -354,34 +354,29 @@ memtx_hash_index_replace(struct index *base, struct tuple *old_tuple,
 	/* HASH index doesn't support ordering. */
 	*successor = NULL;
 
-	if (new_tuple) {
+	if (new_tuple != NULL) {
 		uint32_t h = tuple_hash(new_tuple, base->def->key_def);
 		struct tuple *dup_tuple = NULL;
-		uint32_t pos = light_index_replace(hash_table, h, new_tuple,
-						   &dup_tuple);
+		uint32_t pos = INDEX_OOM_ERRINJ(
+					light_index_replace, light_index_end,
+					hash_table, h, new_tuple, &dup_tuple);
 		if (pos == light_index_end)
-			pos = light_index_insert(hash_table, h, new_tuple);
-
-		ERROR_INJECT(ERRINJ_HASH_INDEX_REPLACE, {
-			light_index_delete(hash_table, pos);
-			pos = light_index_end;
-		});
-
+			pos = INDEX_OOM_ERRINJ(light_index_insert, -1,
```

Comment:
This sets `pos` to -1 on errinj but `light_index_insert` still returns `light_index_end` on oom...

Reply:
Fixed. This by fortune become `light_index_end`)